src/util.js

```diff
@@ -43,6 +43,14 @@ export function dispatch(eventName, { target, cancelable, detail } = {}) {
   return event
 }
 
+export async function nextRepaint() {
+  if (document.visibilityState === "hidden") {
+    await nextEventLoopTick()
+  } else {
+    await nextAnimationFrame()
+  }
+}
```

Comment:
Should this utility return the calls made to the other utilities?

```suggestion
export function nextRepaint() {
  if (document.visibilityState === "hidden") {
    return nextEventLoopTick()
  } else {
    return nextAnimationFrame()
  }
}
```